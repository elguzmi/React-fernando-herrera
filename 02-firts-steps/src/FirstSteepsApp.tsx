import { ItemCounter } from "./shopping-cart/ItemCounter"

export function FirstSteepsApp() {
    return (
        <>
            <h1>Carro de comprar</h1>
            <ItemCounter />
        </>
    )
}