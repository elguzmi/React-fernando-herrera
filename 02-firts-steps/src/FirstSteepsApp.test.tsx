import { render , screen } from "@testing-library/react";
import { describe, test,  expect, vi, afterEach } from "vitest";
import { FirstSteepsApp } from "./FirstSteepsApp";


// eslint-disable-next-line @typescript-eslint/no-unused-vars
const mockItemCounter = vi.fn( (props: unknown)=>{
  return (
    <div data-testid="item-counter" ></div>
  )
})

vi.mock('./shopping-cart/ItemCounter', ()=>({
  ItemCounter : (props : unknown)=> mockItemCounter(props)
}))

// vi.mock('./shopping-cart/ItemCounter', ()=>({
//   ItemCounter : (props : unknown)=> 
//   <div data-testid="item-counter" 
//     name={props.name} 
//     quantity={props.quantity}>
//     </div>
// }));

describe('FirstSteepsApp',()=>{
   //Pasos del ciclo de vida de un test

   afterEach(()=>{
      vi.clearAllMocks(); // Limpiar los mocks
   })

    test('should match snapchot',()=>{
        const {container} = render(<FirstSteepsApp />)
        expect(container).toMatchSnapshot();
        screen.debug();
    })

    test('should render the correct number of item counter components',()=>{
      render(<FirstSteepsApp />)
      screen.debug();
      const itemsCounterComponents = screen.getAllByTestId('item-counter')
      expect(itemsCounterComponents.length).toBe(3);
    })

    test('should render item counter with correct properties',()=>{
      render(<FirstSteepsApp />)

      expect(mockItemCounter).toHaveBeenCalledTimes(3);

      // esos son los parametros que se le pasan al mockItemCounter
      expect(mockItemCounter).toHaveBeenCalledWith(
        {name: 'Nintendo switch 2', quantity: 1}
      );
      expect(mockItemCounter).toHaveBeenCalledWith(
        {name: 'Playstation 5', quantity: 2}
      );
      expect(mockItemCounter).toHaveBeenCalledWith(
        {name: 'Xbox Series X', quantity: 3}
      );
    })
})