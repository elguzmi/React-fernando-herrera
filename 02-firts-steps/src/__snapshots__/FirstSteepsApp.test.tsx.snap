// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstSteepsApp > should match snapchot 1`] = `
<div>
  <h1>
    Carro de comprar
  </h1>
  <div
    data-testid="item-counter"
  />
  <div
    data-testid="item-counter"
  />
  <div
    data-testid="item-counter"
  />
</div>
`;
